NoSuchMethodError: ByteBuffer.clear() on JDK 1.8 After 0.4.0 update

**Describe the bug**
java.lang.NoSuchMethodError: java.nio.ByteBuffer.clear() when running on a machine with JDK 1.8 installed after 0.4.0 update (0.3.3 is working properly)

* Installed JDK Version: 1.8.0_121
```
% java -version 
java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)
```

**To Reproduce**
1. Install jdk 1.8.0
2. Run integrationTest

**Expected behavior**
The test is expected to pass.

**Screenshots**
![스크린샷 2021-10-19 오후 4 38 09](https://user-images.githubusercontent.com/3054645/137864671-577f46a1-fb3a-4fe9-b816-198d5cac3dee.png)

**Android device (please complete the following information):**
 - Device: Galaxy S21 Ultra (SM-G998N)
 - OS: Android 11 One UI 3.1
 - Adam Version: 0.4.0
- Android Debug Server Version: 41

**Additional context**
Full Log (hiding company-related part)
```
java.nio.ByteBuffer.clear()Ljava/nio/ByteBuffer;
java.lang.NoSuchMethodError: java.nio.ByteBuffer.clear()Ljava/nio/ByteBuffer;
	at com.malinskiy.adam.transport.ByteBufferObjectFactory.activateObject(BufferFactory.kt:50)
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:324)
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:223)
	at com.malinskiy.adam.transport.ByteBufferPool.borrow(BufferFactory.kt:42)
	at com.malinskiy.adam.extension.SocketKt.readTransportResponse(Socket.kt:268)
	at com.malinskiy.adam.request.Request.handshake$suspendImpl(Request.kt:47)
	at com.malinskiy.adam.request.Request$handshake$1.invokeSuspend(Request.kt)
	(Coroutine boundary)
	at kotlinx.coroutines.TimeoutKt.withTimeoutOrNull(Timeout.kt:101)
	(Coroutine creation stacktrace)
	at kotlin.coroutines.intrinsics.IntrinsicsKt__IntrinsicsJvmKt.createCoroutineUnintercepted(IntrinsicsJvm.kt:122)
	at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable(Cancellable.kt:30)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at kotlinx.coroutines.debug.junit4.CoroutinesTimeoutStatement$testResult$1.call(CoroutinesTimeoutStatement.kt:23)
	at kotlinx.coroutines.debug.junit4.CoroutinesTimeoutStatement$testResult$1.call(CoroutinesTimeoutStatement.kt:12)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoSuchMethodError: java.nio.ByteBuffer.clear()Ljava/nio/ByteBuffer;
	at com.malinskiy.adam.transport.ByteBufferObjectFactory.activateObject(BufferFactory.kt:50)
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:324)
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:223)
	at com.malinskiy.adam.transport.ByteBufferPool.borrow(BufferFactory.kt:42)
	at com.malinskiy.adam.extension.SocketKt.readTransportResponse(Socket.kt:268)
	at com.malinskiy.adam.request.Request.handshake$suspendImpl(Request.kt:47)
	at com.malinskiy.adam.request.Request$handshake$1.invokeSuspend(Request.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:274)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	... 7 more
```

--- a/adam/src/main/kotlin/com/malinskiy/adam/transport/BufferFactory.kt
+++ b/adam/src/main/kotlin/com/malinskiy/adam/transport/BufferFactory.kt
@@ -17,6 +17,7 @@
 package com.malinskiy.adam.transport
 
 import com.malinskiy.adam.Const
+import com.malinskiy.adam.extension.compatClear
 import org.apache.commons.pool2.PooledObject
 import org.apache.commons.pool2.PooledObjectFactory
 import org.apache.commons.pool2.impl.DefaultPooledObject
@@ -47,14 +48,14 @@ class ByteBufferPool(private val poolSize: Int, private val bufferSize: Int) {
 class ByteBufferObjectFactory(private val bufferSize: Int) : PooledObjectFactory<ByteBuffer> {
     override fun activateObject(p: PooledObject<ByteBuffer>?) {
         p?.`object`?.apply {
-            clear()
+            compatClear()
             order(ByteOrder.BIG_ENDIAN)
         }
     }
 
     override fun destroyObject(p: PooledObject<ByteBuffer>?) {
         p?.`object`?.apply {
-            clear()
+            compatClear()
             order(ByteOrder.BIG_ENDIAN)
         }
     }
@@ -65,7 +66,7 @@ class ByteBufferObjectFactory(private val bufferSize: Int) : PooledObjectFactory
 
     override fun passivateObject(p: PooledObject<ByteBuffer>?) {
         p?.`object`?.apply {
-            clear()
+            compatClear()
             order(ByteOrder.BIG_ENDIAN)
         }
     }

